storage/response_cache.go

```
@@ -90,14 +91,16 @@ func (rc *ResponseCache) GetResponse(cmdID proto.ClientCmdID, reply proto.Respon
 	// If the response is in the cache or we experienced an error, return.
 	rwResp := proto.ReadWriteCmdResponse{}
 	key := keys.ResponseCacheKey(rc.raftID, &cmdID)
-	if ok, err := engine.MVCCGetProto(rc.engine, key, proto.ZeroTimestamp, true, nil, &rwResp); ok || err != nil {
-		if err == nil && rwResp.GetValue() != nil {
-			gogoproto.Merge(reply.(gogoproto.Message), rwResp.GetValue().(gogoproto.Message))
+	ok, err := engine.MVCCGetProto(rc.engine, key, proto.ZeroTimestamp, true, nil, &rwResp)
+	if ok && err == nil {
+		value := rwResp.GetValue()
+		if respProto, isMsg := value.(gogoproto.Message); isMsg {
+			gogoproto.Merge(reply, respProto)
+		} else {
+			panic(fmt.Sprintf("value %s does not implement gogoproto.Message", value))
```

This should probably return an error. If we're just going to panic, we can go back to the old unchecked type assertion.